diff --git a/cmds/boot/stboot/main.go b/cmds/boot/stboot/main.go
index 9a2f01d91e19..9f1ecc071bc5 100644
--- a/cmds/boot/stboot/main.go
+++ b/cmds/boot/stboot/main.go
@@ -34,6 +34,7 @@ var (
 	doDebug = flag.Bool("debug", false, "Print additional debug output")
 	klog    = flag.Bool("klog", false, "Print output to all attached consoles via the kernel log")
 	dryRun  = flag.Bool("dryrun", false, "Do everything except booting the loaded kernel")
+	tlsSkipVerify  = flag.Bool("tlsskipverify", true, "Controls whether a client verifies the server's certificate chain and host name")
 )
 
 var debug = func(string, ...interface{}) {}
diff --git a/cmds/boot/stboot/network.go b/cmds/boot/stboot/network.go
index 6ea0b7162826..76fe69911f1d 100644
--- a/cmds/boot/stboot/network.go
+++ b/cmds/boot/stboot/network.go
@@ -158,8 +158,12 @@ func findNetworkInterfaces() ([]netlink.Link, error) {
 func download(url *url.URL, httpsRoots *x509.CertPool) ([]byte, error) {
 	// setup client with values taken from http.DefaultTransport + RootCAs
 	tls := &tls.Config{
-		RootCAs: httpsRoots,
+		InsecureSkipVerify: *tlsSkipVerify,
 	}
+	if !*tlsSkipVerify {
+		tls.RootCAs = httpsRoots
+	}
+
 	client := http.Client{
 		Transport: (&http.Transport{
 			Proxy: http.ProxyFromEnvironment,
